internal/appsec/listener/grpcsec/grpc.go

```diff
@@ -137,9 +146,11 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
 	}
 
 	wafResult := shared.RunWAF(wafCtx, waf.RunAddressData{Persistent: values})
-	if wafResult.HasActions() || wafResult.HasEvents() {
+	if wafResult.HasEvents() {
+		addEvents(wafResult.Events)
+	}
+	if wafResult.HasActions() {
 		interrupt := shared.ProcessActions(op, wafResult.Actions)
-		shared.AddSecurityEvents(&op.SecurityEventsHolder, l.limiter, wafResult.Events)
 		log.Debug("appsec: WAF detected an attack before executing the request")
```

Comment:
Same here, the log needs to go up on line 150

Reply:
fixed - thanks